### DISCUSSION TITLE
F-Sy-H: Enabling chromas?

### DISCUSSION THREAD

**Discussion by cinerea0** [z-shell/zi#174] [Q&A] [answered]
I'd like to know how to enable non-default chromas, and I believe this should be explicitly documented as well. In my case, I saw that the chroma for git is not active by default, and I'd like to enable it.

**Comment by ss-o** [comment, marked as answer]
> I'd like to know how to enable non-default chromas, and I believe this should be explicitly documented as well. In my case, I saw that the chroma for git is not active by default, and I'd like to enable it.

Have you checked [chromas guide](https://z.digitalclouds.dev/ecosystem/plugins/f-sy-h#chroma-guide-for-f-sy-h)? :memo:

**Reply by cinerea0** [reply]
I have, and it does not state how to enable an individual chroma.

**Reply by ss-o** [reply]
> I have, and it does not state how to enable an individual chroma.

Chromas are enabled [here](https://github.com/z-shell/F-Sy-H/blob/99fddaa6c77776669c5038fad8ef8badf2f159b9/fast-highlight#L172-L276) and specifically git is [here](https://github.com/z-shell/F-Sy-H/blob/99fddaa6c77776669c5038fad8ef8badf2f159b9/fast-highlight#L190) and loaded files of chromas are [here](https://github.com/z-shell/F-Sy-H/blob/99fddaa6c77776669c5038fad8ef8badf2f159b9/F-Sy-H.plugin.zsh#L337).

So basically need to load the file and enable it accordingly.

**Reply by cinerea0** [reply]
If I'm understanding those links correctly, while `-git.ch` isn't loaded in fast-highlight because `chroma-git` refers to `main-chroma.ch%git`, it still gets loaded by the plugin file because it's in the list of chromas there. Is that correct? If so, why are git subcommands not being colored for me? You can see an example of what I mean here: https://asciinema.org/a/ilFEFDcJOixx8X8X3rqAk3A7y

**Reply by ss-o** [reply]
Correct. What theme are you using? (`fast-theme -s`). I am not sure but it looks like highlighting works, but the colors are incorrect. You can try to reset or swap themes see: `fast-theme -h`. If it doesn't help, then you need to provide exact details to recreate your environment to debug the issue (can setup at: `https://github.com/z-shell/playground`).

**Reply by cinerea0** [reply]
I'm using the default theme; neither switching nor resetting the theme helps. I'm going to try cleaning all of F-Sy-H out and reinstalling to see if that will help. Are there any files besides the repo itself and the cache directory I should take care of?

**Reply by ss-o** [reply]
[![asciicast](https://asciinema.org/a/Fuhd9durFmMyS0HjEEBzqbPFN.svg)](https://asciinema.org/a/Fuhd9durFmMyS0HjEEBzqbPFN)

**Reply by ss-o** [reply]
Try in [docker](https://github.com/z-shell/zd/pkgs/container/zd) e.g.:

```sh
docker run --rm -it ghcr.io/z-shell/zd:latest
```

**Reply by cinerea0** [reply]
I did some more testing, and I discovered that the convoluted, system-wide installation method I was using previously must have been the cause of the trouble. Everything works perfectly using antigen or ZI. Thank you so much for helping me troubleshoot this!
